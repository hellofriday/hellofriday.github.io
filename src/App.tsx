import React from 'react';
import {Route, Routes} from "react-router-dom";
import FoodList from "./pages/foodList";

function App() {
  return (
      <Routes>
        <Route path="/" element={<FoodList />}/>
      </Routes>
  );
}

export default App;
