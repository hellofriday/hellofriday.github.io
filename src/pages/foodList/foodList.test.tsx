import { fireEvent, render, screen } from "@testing-library/react";
import FoodList from "./foodList";
import {BrowserRouter} from "react-router-dom";
import App from "../../App";


describe("FoodList", () => {
    it('should render correctly', () => {
        const { container } = render(<FoodList />, {wrapper: BrowserRouter});
        expect(container.querySelector('.container')).toBeInTheDocument()
    });

    it("should show/hide '去结账' button correctly when click food add button or minus button", () => {
        const { container } = render(<FoodList />, {wrapper: BrowserRouter});
        const firstFood = container.querySelector('.item')! as HTMLElement
        expect(firstFood).toMatchSnapshot()

        fireEvent.click(firstFood.querySelector('.add')!);
        expect(container.querySelector('.orderBtn')).toBeInTheDocument()

        fireEvent.click(firstFood.querySelector('.minus-btn')!);
        expect(container.querySelector('.orderBtn')).not.toBeInTheDocument()
    });

    it("should navigate to order page when click '去结账' button", () => {
        const { container } = render(<App />, {wrapper: BrowserRouter});
        const firstFood = container.querySelector('.item')! as HTMLElement
        fireEvent.click(firstFood.querySelector('.add')!);
        const orderBtn = container.querySelector('.orderBtn')! as HTMLElement
        expect(orderBtn).toBeInTheDocument()

        fireEvent.click(orderBtn);
        expect(screen.getByText('请结账')).toBeInTheDocument();
    });
});
