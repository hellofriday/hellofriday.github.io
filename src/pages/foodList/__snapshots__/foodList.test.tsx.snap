// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoodList should show/hide '去结账' button correctly when click food add button or minus button 1`] = `
<div
  class="item"
>
  <span
    class=" lazy-load-image-background opacity"
    style="color: transparent; display: inline-block;"
  >
    <img
      alt="cake"
      class="img"
      src="cake.jpg"
    />
  </span>
  <div
    class="info"
  >
    <div
      class="name"
    >
      蛋糕0
    </div>
    <div
      class="price"
    >
      <b>
        ￥
      </b>
      <span>
        10
      </span>
    </div>
  </div>
  <div
    class="gap"
  />
  <div
    class="add"
  />
</div>
`;
